### DISCUSSION TITLE
URGENT: Installation issue

### DISCUSSION THREAD

**Discussion by backpack-operations** [Laravel-Backpack/community-forum#808] [Q&A (Help)]
Hello Team,

I just received this on email:
"I am facing some issues during the installation <img width="959" alt="image" src="https://github.com/Laravel-Backpack/community-forum/assets/144793034/8222141e-09a3-4289-a4a4-130e8414d017"> ,also, I have added a button, it is showing locally, but online it is not I don't know it is it from the composer or not
please assist it is too urgent

"

**Comment by pxpm** [comment]
Hello @backpack-operations 

I think the problem relies on the fact that composer is trying to get the package from github and not from our private repository. 

Please advise the user to add into their `composer.json` the following entry:
```json
"repositories": {
        "backpack": {
            "type": "composer",
            "url": "https://repo.backpackforlaravel.com/"
        }
    },
```

If that does not solve it, or the user already has that in composer.json, please ask for a copy of his `composer.json` file so that we can debug it. 

Cheers

**Comment by rawad1** [comment]
[it]([composer.json](https://github.com/Laravel-Backpack/community-forum/files/13976477/composer.json)) didn't work.
this is the composer.json


  GNU nano 2.3.1                                                       File: composer.json

{
    "name": "laravel/laravel",
    "type": "project",
    "description": "The skeleton application for the Laravel framework.",
    "keywords": [
        "laravel",
        "framework"
    ],
    "license": "MIT",
        "require": {
        "php": "^8.1",
        "backpack/crud": "^6.0",
        "backpack/devtools": "*",
        "backpack/editable-columns": "*",
        "backpack/pro": "^2.0",
        "backpack/theme-coreuiv2": "^1.1",
        "backpack/theme-coreuiv4": "*",
        "guzzlehttp/guzzle": "^7.2",
        "laracasts/flash": "*",
        "laravel/framework": "^10.10",
        "laravel/sanctum": "^3.2",
        "laravel/tinker": "^2.8",
        "predis/predis": "*"
    },
    "require-dev": {
        "backpack/generators": "^4.0",
        "fakerphp/faker": "^1.9.1",
        "laravel/pint": "^1.0",
        "laravel/sail": "^1.18",
        "mockery/mockery": "^1.4.4",
        "nunomaduro/collision": "^7.0",
        "phpunit/phpunit": "^10.1",
        "spatie/laravel-ignition": "^2.0"
    },

    "autoload": {
        "psr-4": {
            "App\\": "app/",
            "Database\\Factories\\": "database/factories/",
            "Database\\Seeders\\": "database/seeders/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
    "scripts": {
        "post-autoload-dump": [
            "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
            "@php artisan package:discover --ansi"
        ],
        "post-update-cmd": [
            "@php artisan vendor:publish --tag=laravel-assets --ansi --force"
        ],
        "post-root-package-install": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ],
        "post-create-project-cmd": [
            "@php artisan key:generate --ansi"
        ],
        "post-install-cmd": [
            "php artisan storage:link"
        ]
    },
    "extra": {
        "laravel": {
            "dont-discover": []
        }
    },
    "config": {
        "optimize-autoloader": true,
        "preferred-install": "dist",
        "sort-packages": true,
        "allow-plugins": {
            "pestphp/pest-plugin": true,
            "php-http/discovery": true
        }
    },
    "repositories": {
        "backpack": {
            "type": "composer",
            "url": "https://repo.backpackforlaravel.com/"
        }
    },
    "minimum-stability": "stable",
    "prefer-stable": true
}

**Reply by tabacitu** [reply]
Thank you @rawad1 - your composer.json looks fine to me. 

The screenshot you pasted above shows me Composer has failed to download from `dist`, so it's trying to download from `source`. Which you cannot do, as [written in our docs](https://backpackforlaravel.com/docs/6.x/faq#errors-when-installing-paid-add-ons-1). So the root problem should be right above that, it shouldn't fail to download from `dist` in the first place. **Can you please provide a bigger screenshot of the command line, when you get the error?** The actual problem should be above. 

Something like "_failed to download from dist, getting error 429_"

Cheers!

**Reply by tabacitu** [reply]
I'm pretty sure it will say Error 402. Which means "Payment required". 

It looks to me like you're trying to download DevTools and EditableColumns, but you have not purchased those products, you have only purchased PRO. To have access them, please purchase them on our website.

**Comment by rawad1** [comment]
the requested changes already done, and I have added the credentials, but still the same issue

**Comment by rawad1** [comment]
this is the full screenshot
[image: image.png]
best regards,

Rawad Alzghayar

Founder

RM

**Reply by tabacitu** [reply]
The screenshot isn't visible on Github. Please open Github and post it there, so we can see it. Also please see my answers here - https://github.com/Laravel-Backpack/community-forum/discussions/808#discussioncomment-8168725 - I think they get to the bottom of your problem.

**Comment by rawad1** [comment]
Yesterday I purchased the package
the issue that I am also facing besides the composer is the "New Export"
button which is showing locally but when I deploy the code online it is not
showing

Local:
[image: image.png]

online
[image: image.png]

best regards,

Rawad Alzghayar

Founder

RM

**Reply by tabacitu** [reply]
For different issues, please open different threads here: https://github.com/Laravel-Backpack/community-forum/discussions 

Make sure you add your issue on Github, not by email. Otherwise we won't be able to see the images you post.

**Comment by pxpm** [comment]
There are multiple issues here in this thread. There are multiple threads for this issue.

I've replied with more information here: https://github.com/Laravel-Backpack/community-forum/discussions/810#discussioncomment-8168755

Closing this one. 

Thanks for the report @backpack-operations 🙏
